### PR TITLE
Mepts 937 tx pvls by source

### PR DESCRIPTION
**Pull request by ningosi**
A new reports needs to be added in Reporting Module with the name “Tx_PVLS By Source”.

It contains two tabs, one is “TX_PVLS_Lab_FSR” and another is TX_PVLS_Ficha_Mestre”. The first one should consider only Laboratory form and FSR form as sources for Viral Load and the second one should consider only Ficha Mestre (Ficha Clinica and Ficha Resumo) as Viral Load sources. Both Denominators and Numerators.

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxPvlsBySourceQueries.java [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    map.put("6", String.valueOf(adultSeguimentoEncounter));
+    map.put("9", String.valueOf(pediatriaSeguimentoEncounter));
+    map.put("53", String.valueOf(mastercardEncounter));
+    map.put("856", String.valueOf(vlConceptQuestion));
+    map.put("1305", String.valueOf(vlQualitativeConceptQuestion));
+    String query =
+        "SELECT fn1.patient_id FROM( "
+            + " SELECT  patient_id, MAX(data_carga) AS data_carga FROM( "
+            + " SELECT patient_id,data_carga FROM "
+            + " (SELECT p.patient_id,MAX(o.obs_datetime) data_carga "
```
Comment (eliazmuchave) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxPvlsBySourceQueries.java:146]: Please check against the spec if should be encounter_datetime or obs_datetime.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxPvlsBySourceC/TxPvlsClinicalOrFichaResumoCohortQueries.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package org.openmrs.module.eptsreports.reporting.library.cohorts.TxPvlsBySourceC;
+
+import org.openmrs.module.eptsreports.metadata.HivMetadata;
+import org.openmrs.module.eptsreports.reporting.library.cohorts.TxPvlsCohortQueries;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TxPvlsClinicalOrFichaResumoCohortQueries {
```
Comment (eliazmuchave) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxPvlsBySourceC/TxPvlsClinicalOrFichaResumoCohortQueries.java]: This class is not being used

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxPvlsBySourceQueries.java (hunk 1 above)
Comment (psmatsinhe) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxPvlsBySourceQueries.java:146]: @ningosi can you please check this, I believe this should be encounter_datetime

**Review by psmatsinhe — CHANGES_REQUESTED** [review]
@ningosi Please have a look

**Comment by ningosi** [comment]
@psmatsinhe and @eliazmuchave I am reusing the Pvls code as it is, but for this report I alter encounter types. In our previous tests of pvls and which is certified, we found out that approach was correct. I just let most of the code the way they are and only concentrated in supplying and altering the encounter types. Maybe we can leave it the way it is and get validations after BA review.

**Review by psmatsinhe — APPROVED** [review]
Looks good to me

**Pull request merged**